### PR TITLE
grid: migrate upstream changes for compatibility with 417 clay + kiln 

### PR DESCRIPTION
**Pull request by tomholford**
# Context

This resolves https://github.com/tloncorp/landscape-apps/issues/1271 by migrating the frontend [changes](https://github.com/urbit/urbit/compare/master...next/417) made in the urbit/urbit repo to support the [kelvin 417 clay + kiln changes](https://roadmap.urbit.org/project/agents-in-clay).

# Changes

- import commits from urbit/urbit to integrate this repo with the clay + kiln changes for arvo 417
- replace usage of Vats with Pikes
- breaking: remove rendered `aeon` from system update view (the value was previously exposed via Vat, no longer exposed via Pike)

# Preview

## After

image

image

# TODO

- [x] resolve remaining merge conflicts
- [x] publish `@urbit/api` package and point the package.json to updated version

### CODE REVIEW

**Comment by jamesacklin** [comment]
Looks good. I can't tell from the diff, but have we decremented the kelvin version in the docket as well?

**Review by patosullivan — APPROVED** [review]
Nice, code looks good and runs well locally.

File: ui/src/preferences/about-system/AboutSystem.tsx [hunk 1]
```diff
@@ -45,15 +43,6 @@ export const AboutSystem = () => {
...
             </p>
-            {systemBlocked && (
-              <p>
-                Aeon {aeon}{' '}
-                <span className="text-orange-500 mx-4 space-x-2">
-                  <span>&mdash;&gt;</span> <span>/</span> <span>&mdash;&gt;</span>
-                </span>{' '}
-                Aeon {nextAeon}
-              </p>
-            )}
```
Comment (arthyn) [inline ui/src/preferences/about-system/AboutSystem.tsx:56]: is there no replacement for this in the pike system?

File: ui/src/preferences/about-system/UpdatePreferences.tsx [hunk 2]
```diff
@@ -1,81 +1,23 @@
...
+
+export const UpdatePreferences = () => {
+  const desk = 'base';
+  const appName = 'your Urbit';
+  const pike = usePike(desk);
+  const srcShip = pike?.sync?.ship;
+  const { toggleInstall } = useKilnState();
+  
   return (
-    <div className="inner-section space-y-8 relative">
```
Comment (arthyn) [inline ui/src/preferences/about-system/UpdatePreferences.tsx:48]: This should wrap around the system updates source setter. Also not sure about the title changing here @urcades

**Review by arthyn — CHANGES_REQUESTED** [review]
One small change and one question

File: ui/src/preferences/about-system/AboutSystem.tsx (hunk 1 above)
Comment (tomholford) [reply ui/src/preferences/about-system/AboutSystem.tsx:56]: Here is the schema for Pike:
```ts
export interface Pike {
  /**
   * Hash of the desk, rendered as `@uv`
   *
   * @remarks
   * Equivalent to
   * ```hoon
   * .^(@uv %cz /=desk=)
   * ```
   */
  hash: string;
  sync: {
    /**
     * Source desk for this Pike
     */
    desk: string;
    /**
     * Source ship for this Pike
     */
    ship: string;
  } | null;
  /**
   *  {@link Weft}s associated with this Pike
   */
  wefts: Weft[];
  /**
   * how live is this pike?
   * 
   * live - app is running
   * held - app is not running, but is trying to run. this state can be entered
   * in two main ways: 
   *   - when installing an app but it hasn't finished downloading (or it did 
   *     but failed to install for some reason)
   *   - when user forced a kelvin upgrade by suspending desks.
   * dead - app is not running
   */
  zest: "live" | "dead" | "held";
}
```

And here's a Weft:
```ts
/**
 * A component's kelvin version
 */
export interface Weft {
  /**
   *  Name of the component
   *
   *  @remarks
   *  Usually %zuse, %hoon, or %lull
   */
  name: string;
  /**
   * Kelvin version
   *
   */
  kelvin: number;
}
```

Comment (tomholford) [reply ui/src/preferences/about-system/AboutSystem.tsx:56]: So looking at the new schema posted in previous comment, not seeing any `aeon` fields.

File: ui/src/preferences/about-system/UpdatePreferences.tsx (hunk 2 above)
Comment (tomholford) [reply ui/src/preferences/about-system/UpdatePreferences.tsx:48]: Good call, will make the change

Comment (tomholford) [reply ui/src/preferences/about-system/UpdatePreferences.tsx:48]: Added to both instances of `SourceSetter`, looks much better:

image

image

**Comment by tomholford** [comment]
@arthyn Addressed feedback on the `SourceSetter` container. As for `aeon`, posted the type corresponding to what is returned by the backend to confirm that the property is not being returned.

**Review by arthyn — APPROVED** [review]
lgtm!

**Pull request merged**